Tools/CISupport/ews-build/steps.py

```diff
@@ -169,30 +169,19 @@ def get_pr_json(self, pr_number, repository_url=None):
             return None
         return pr_json
 
-    def _is_pr_closed(self, pr_number, repository_url=None):
-        if not pr_number:
-            self._addToLog('stdio', 'Skipping pull request status validation since pull request is None.\n')
-            return -1
-
-        pr_json = self.get_pr_json(pr_number, repository_url)
+    def _is_pr_closed(self, pr_json):
         if not pr_json or not pr_json.get('state'):
-            self._addToLog('stdio', 'Unable to fetch pull request {}.\n'.format(pr_number))
+            self._addToLog('stdio', 'Cannot determined closed status, no pull request details provided.\n')
             return -1
         if pr_json.get('state') in self.pr_closed_states:
             return 1
         return 0
 
-    def _is_pr_obsolete(self, pr_number, repository_url=None):
-        pr_json = self.get_pr_json(pr_number, repository_url)
-        if not pr_json:
-            self._addToLog('stdio', 'Unable to fetch pull request {}.\n'.format(pr_number))
-            return -1
-
+    def _is_pr_obsolete(self, pr_json):
         pr_sha = pr_json.get('head', {}).get('sha', '')
-        if not pr_sha:
-            self._addToLog('stdio', 'Failed to fetch hash of pull request {}\n'.format(pr_number))
+        if not pr_json or not pr_sha:
+            self._addToLog('stdio', 'Cannot determined if pull request is obsolete, no details provided.\n')
```

Comment:
Ditto. Also is it the PR which is obsolete or the hash which is obsolete? In patch workflow, bug is not obsolete, it's the patch which is obsolete.

Can frame it something like:
'Cannot determined if hash {} is obsolete' or 'Cannot determined if pull request is obsolete'.